### PR TITLE
[FEAT] Cooking Skills

### PR DESCRIPTION
**Pull request by 0xSacul**
# Description

This PR adds the Cooking skills from skill revamp.

# What needs to be tested by the reviewer?

Please describe how this can be tested.

# Checklist:

- [ ] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [ ] I have read the contributing guidelines and agree to the T&Cs
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/game/events/landExpansion/collectRecipe.ts [hunk 1]
```diff
@@ -17,6 +18,38 @@ type Options = {
...
+  if (bumpkin.skills["Double Nom"]) {
+    const doubleNomChance = randomInt(1, 100);
+    if (doubleNomChance <= 50) {
+      total = total.mul(2);
+    }
+  }
+
+  // Firey Jackpot - 20% chance to double the amount of food collected from Fire Pit
+  if (
+    bumpkin.skills["Firey Jackpot"] &&
```
Comment (brian-man) [inline src/features/game/events/landExpansion/collectRecipe.ts:43]: The correct spelling is "Fiery".

File: src/features/game/types/game.ts [hunk 2]
```diff
@@ -540,6 +540,7 @@ export type Mine = Position;
 export type BuildingProduct = {
   name: CookableName;
   readyAt: number;
+  amount: Decimal;
```
Comment (adamhannigan) [inline src/features/game/types/game.ts:543]: I would consider keeping this as a `number`.

There is some strange behaviour when persisting `Decimal` types - unless absolutely needed, we opt for storing just as number.

Also, I would suggest making this an optional value so it is backwards compatible with all of the current recipes that are already cooking.

`amount?: number`

**Review by adamhannigan — COMMENTED** [review]
Nice work mate - just a suggestion around the data structure 👌

**Review by adamhannigan — APPROVED** [review]
LGTM

**Pull request merged**